### PR TITLE
Fix issue #17902

### PR DESCRIPTION
**Pull request by paosidufygthrj**

### CODE REVIEW

File: cocos/2d/CCLabel.cpp [hunk 1]
```diff
@@ -416,7 +416,12 @@ Label::Label(TextHAlignment hAlignment /* = TextHAlignment::LEFT */,
         if (_fontAtlas && _currentLabelType == LabelType::TTF && event->getUserData() == _fontAtlas)
         {
             _fontAtlas = nullptr;
+            auto lineHeight = _lineHeight;
```
Comment (zilongshanren) [inline cocos/2d/CCLabel.cpp:419]: why declare a temp variable here? I think we could just use `_lineHeight` here.

File: tests/cpp-tests/Classes/LabelTest/LabelTestNew.cpp [hunk 2]
```diff
@@ -3397,4 +3398,34 @@ std::string LabelIssueLineGap::subtitle() const
...
+}
+
+void LabelIssue17902::purgeCachedData()
+{
+    FontAtlasCache::purgeCachedData();
+}
+
+std::string LabelIssue17902::title() const
+{
+    return "Github Issue XXX";
```
Comment (zilongshanren) [inline tests/cpp-tests/Classes/LabelTest/LabelTestNew.cpp]: please change `XXX` to `17902`, thanks.

**Comment by zilongshanren** [comment]
@paosidufygthrj 
Thanks, I have commented on this PR.

**Review by zilongshanren — APPROVED** [review]
This PR looks good to me.

File: tests/cpp-tests/Classes/LabelTest/LabelTestNew.cpp (hunk 2 above)
Comment (paosidufygthrj) [reply tests/cpp-tests/Classes/LabelTest/LabelTestNew.cpp]: Fixed. I will be more careful in the future.

File: cocos/2d/CCLabel.cpp (hunk 1 above)
Comment (paosidufygthrj) [reply cocos/2d/CCLabel.cpp:419]: Because call setTTFConfig reset _lineHeight .

**Pull request merged**